bot/exts/moderation/incidents.py

```diff
@@ -402,10 +492,104 @@ async def on_raw_reaction_add(self, payload: discord.RawReactionActionEvent) ->
+
+        If the message edit (`msg_after`) is a incident then run it through `extract_message_links`
+        to get all the message link embeds (embeds which contain information about that particular
+        link), this message link embeds are then sent into the channel.
+
+        The edited message is also passed into `add_signals` if it is a incident message.
+        """
+        if is_incident(msg_after):
+            webhook_embed_list = await extract_message_links(msg_after, self.bot)
+            webhook_msg_id = await self.message_link_embeds_cache.get(msg_before.id)
+
+            if webhook_msg_id:
+                await self.incidents_webhook.edit_message(
+                    message_id=webhook_msg_id,
+                    embeds=[x for x in webhook_embed_list if x is not None],
+                )
+                return
+
+            await self.send_webhooks(webhook_embed_list, msg_after, self.incidents_webhook)
+
+    @Cog.listener()
+    async def on_message_delete(self, message: discord.Message) -> None:
+        """
+        Delete message link embeds for `message`.
+
+        Search through the cache for message, if found delete it from cache and channel.
+        """
+        if is_incident(message):
+            await self.delete_msg_link_embed(message)
+
+    async def send_webhooks(
+            self,
+            webhook_embed_list: t.List,
+            message: discord.Message,
+            webhook: discord.Webhook,
+    ) -> t.Optional[int]:
+        """
+        Send Message Link Embeds to #incidents channel.
+
+        Uses the `webhook` passed in as a parameter to send
+        the embeds in the `webhook_embed_list` parameter.
+
+        After sending each webhook it maps the `message.id`
+        to the `webhook_msg_ids` IDs in the async redis-cache.
+        """
+        try:
+            webhook_msg = await webhook.send(
+                embeds=[x for x in webhook_embed_list if x is not None],
+                username=sub_clyde(message.author.name),
+                avatar_url=message.author.avatar_url,
+                wait=True,
+            )
+            log.trace("Message Link Embed sent successfully.")
+
+        except discord.DiscordException:
+            log.exception(
+                f"Failed to send message link embed {message.id} to #incidents."
+            )
+
+        else:
+            await self.message_link_embeds_cache.set(message.id, webhook_msg.id)
+            log.trace("Message Link Embed Sent successfully!")
+            return webhook_msg.id
+
+    async def delete_msg_link_embed(self, message: discord.Message) -> None:
+        """Delete discord message link message found in cache for `message`."""
+        log.trace("Deleting discord links webhook message.")
+        webhook_msg_id = await self.message_link_embeds_cache.get(message.id)
+
+        if webhook_msg_id:
+            await self.incidents_webhook.delete_message(webhook_msg_id)
```

Comment:
This raises `discord.errors.NotFound` whenever a webhook message is deleted, not sure why. Also this doesn't account for a case where the webhook message was already deleted for some reason.

Reply:
This doesn't raise an error anymore, but it still fires the "skipping" log message whenever I action an incident, even if the embed is still there.

Reply:
I couldn't replicate this on my test bot, could you post a screenshot of it?